File: StudentCredit.Application/Credits/Dtos/SearchCreditFilter.cs
using StudentCredit.Data.Applications.Common.Enums;
using StudentCredit.Data.Common.Enums;

namespace StudentCredit.Application.Credits.Dtos
{
    public class SearchCreditFilter
	{
		public int? BankId { get; set; }

		public string Uin { get; set; }

		public string CreditNumber { get; set; }

		public string Uan { get; set; }

		public DateTime? ContractDate { get; set; }

		public CreditType? Type { get; set; }

		public int? InstitutionId { get; set; }

		public string StudentFullName { get; set; }

		public CommitState CommitState { get; set; }

		public int Limit { get; set; } = 10;
		public int Offset { get; set; } = 0;
	}
}

File: WebApplication/src/modules/application/credit/services/search-credit.filter.ts
import { Injectable } from '@angular/core';
import { NomenclatureDto } from 'src/infrastructure/models/nomenclature.dto';
import { BaseSearchFilter } from 'src/infrastructure/services/base-search.filter';
import { CreditType } from '../../common/enums/credit-type.enum';
import { CommitState } from '../../common/enums/commit-state.enum';

@Injectable()
export class SearchCreditFilter extends BaseSearchFilter {
  bank: NomenclatureDto;
  bankId: number | null;
  uin: string;
  creditNumber: string;
  uan: string;
  contractDate: Date | null;
  type: CreditType | null = null;
  institution: NomenclatureDto;
  institutionId: number | null;
  studentFullName: string;
  commitState: CommitState = CommitState.approved;

  constructor() {
    super(10);
  }
}
